src/losses/functions.jl

```diff
@@ -46,6 +46,41 @@ function huber_loss(ŷ, y; agg=mean, δ=ofeltype(ŷ, 1))
    agg(((abs_error.^2) .* temp) .* x .+ δ*(abs_error .- x*δ) .* (1 .- temp))
 end
 
+"""
+    label_smoothing(y::Union{Number, AbstractArray}, α; dims=1)
+
+Returns smoothed labels, meaning the confidence on label values are relaxed.
+
+when `y` given as one-hot vector or batch of one-hot, its calculated as
+
+    Label smoothing = y .* (1 - α) .+ α * 1 // size(y, 1)
+
+when `y` given as a number or batch of numbers for binary classification,
+its calculated as
+
+    Label smoothing = y .* (1 - α) .+ α*1//2
```

Comment:
```suggestion
    y .* (1 - α) .+ α / 2
```
Let's not be too pedantic with type problems in the docstring